File: core/src/com/unciv/models/gamebasics/TileResource.java
package com.unciv.models.gamebasics;

import com.unciv.models.stats.NamedStats;
import com.unciv.models.stats.FullStats;

import java.util.Collection;

public class TileResource extends NamedStats implements ICivilopedia {
    public ResourceType resourceType;
    public Collection<String> terrainsCanBeFoundOn;
    public String improvement;
    public FullStats improvementStats;

    /// <summary>
/// The building that improves this resource, if any. E.G.: Granary for wheat, Stable for cattle.
/// </summary>
    public String building;
    public Building GetBuilding(){return building ==null ? null : GameBasics.Buildings.get(building);}

    public String revealedBy;

    @Override
    public String getDescription() {
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append(new FullStats(this)+"\r\n");
        stringBuilder.append("Can be found on " + com.unciv.models.gamebasics.StringUtils.join(", ", terrainsCanBeFoundOn));
        stringBuilder.append("\r\n\r\nImproved by "+ improvement +"\r\n");
        stringBuilder.append("\r\nBonus stats for improvement: "+ improvementStats +"\r\n");
        return stringBuilder.toString();
    }
}

File: core/src/com/unciv/models/gamebasics/TileResource.kt
package com.unciv.models.gamebasics

import com.unciv.models.linq.Linq
import com.unciv.models.stats.NamedStats
import com.unciv.models.stats.Stats

class TileResource : NamedStats(), ICivilopedia {
    override val description: String
        get(){
            val stringBuilder = StringBuilder()
            stringBuilder.appendln(this.clone().toString())
            stringBuilder.appendln("Can be found on " + terrainsCanBeFoundOn.joinToString())
            stringBuilder.appendln()
            stringBuilder.appendln("Improved by " + improvement)
            stringBuilder.appendln("Bonus stats for improvement: " + improvementStats)
            return stringBuilder.toString()
        }

    @JvmField var resourceType: ResourceType = ResourceType.Bonus
    @JvmField var terrainsCanBeFoundOn: Linq<String> = Linq()
    @JvmField var improvement: String? = null
    @JvmField var improvementStats: Stats? = null

    /**
     * The building that improves this resource, if any. E.G.: Granary for wheat, Stable for cattle.
     */
    @JvmField var building: String? = null
    @JvmField var revealedBy: String? = null

    fun GetBuilding(): Building? {
        return if (building == null) null else GameBasics.Buildings[building]
    }
}

